File: agent-training-management-api/Models/Entities/Skill.cs
namespace AtmAPI.Models.Entities;

public class Skill : EntityBase
{
	[MaxLength(ModelConstants.Commons.MaxLength)]
	public string Name { get; set; } = default!;

	[MaxLength(ModelConstants.Commons.DescriptionLength)]
	public string? Description { get; set; }
	public int CategoryId { get; set; } = default!;

	public virtual Category Category { get; set; } = default!;
	public virtual ICollection<Class>? Classes { get; set; }
}

File: agent-training-management-ui/src/types/entities/skill.ts
import type { EntityBase } from '@/commons';

import type { CategoryResponse } from './category';

export interface SkillUpsert {
	name?: string;
	description?: string;
}

export interface SkillResponse extends SkillUpsert, EntityBase {
	category?: CategoryResponse;
}

export interface SkillReportUpdateVerifiedRequest {
	score?: number;
	class_id?: number | string;
}
